pkg/cmd/infra/router/template_test.go

```diff
@@ -0,0 +1,132 @@
+				{Name: "Accept", Value: "'text/plain, text/html'", Action: "Set"},
+				{Name: "Accept-Encoding", Action: "Delete"},
+			},
+			expectError: false,
+		},
+		{
+			description: "should percent decode the multiple header values with simple non encoded strings",
+			inputValue:  "X-Frame-Options:DENY:Set",
+			expectedValue: []templateplugin.HTTPHeader{
+				{Name: "X-Frame-Options", Value: "'DENY'", Action: "Set"},
+			},
+			expectError: false,
+		},
+		{
+			description:        "When Action is `Set` it should not percent decode the incorrectly encoded value part Eg: percent is not followed by two hexadecimal digits",
+			inputValue:         "x-forwarded-client-cert:%25%7B%2BQ%7D%5Bssl_c_der%2Cbase64%5:Set",
+			expectError:        true,
+			expectedValue:      nil,
+			expectErrorMessage: "failed to decode percent encoding",
+		},
+		{
+			description:        "Invalid Action when header input spec has 2 parts",
+			inputValue:         "Accept-Encoding:Delete%2",
+			expectError:        true,
+			expectedValue:      nil,
+			expectErrorMessage: "invalid action Delete%2",
+		},
+		{
+			description:        "Invalid Action when header input spec has 3 parts",
+			inputValue:         "X-Frame-Options:DENY:Bad",
+			expectedValue:      nil,
+			expectError:        true,
+			expectErrorMessage: "invalid action Bad",
+		},
+		{
+			description:        "Should not allow specifying headers with no action or value",
+			inputValue:         "X-Frame-Options",
+			expectError:        true,
+			expectErrorMessage: "invalid HTTP header input specification: ",
+		},
+		{
+			description:        "Should not allow blank values for headers",
+			inputValue:         "",
+			expectedValue:      nil,
+			expectError:        true,
+			expectErrorMessage: "encoded header string not present",
+		},
+		{
+			description:        "Should fail with incorrect header name when HTTP header input specification has 3 parts",
+			inputValue:         "X-Frame-Options**!=:DENY:Set",
+			expectedValue:      nil,
+			expectError:        true,
+			expectErrorMessage: "invalid HTTP header name: X-Frame-Options**!",
+		},
+		{
+			description:        "Should fail with incorrect header name when HTTP header input specification has 2 parts",
+			inputValue:         "X-Frame-Options**!=:Delete",
+			expectedValue:      nil,
+			expectError:        true,
+			expectErrorMessage: "invalid HTTP header name: X-Frame-Options**!",
+		},
+		{
+			description:        "Should fail when header spec consists of more than 3 parts than Name:Value:Action",
+			inputValue:         "X-Frame-Options:DENY:Set:Invalid",
+			expectedValue:      nil,
+			expectError:        true,
+			expectErrorMessage: "invalid HTTP header input specification: X-Frame-Options:DENY:Set:Invalid",
+		},
+	}
+
+	for _, tc := range testCases {
+		switch gotValue, err := parseHeadersToBeSetOrDeleted(tc.inputValue); {
+		case !cmp.Equal(gotValue, tc.expectedValue):
+			t.Errorf(" expected %s, got %s", tc.expectedValue, gotValue)
```

Comment:
```suggestion
		gotValue, err := parseHeadersToBeSetOrDeleted(tc.inputValue); 
```

Reply:
I am not sure what you want on this. We need switch.